File: OsmAnd/src/net/osmand/plus/track/helpers/GpxDirItem.java
package net.osmand.plus.track.helpers;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import net.osmand.gpx.GpxParameter;
import net.osmand.plus.OsmandApplication;

import java.io.File;

public class GpxDirItem extends DataItem {

	public GpxDirItem(@NonNull OsmandApplication app, @NonNull File file) {
		super(app, file);
	}

	@Override
	public boolean isValidValue(@NonNull GpxParameter parameter, @Nullable Object value) {
		return parameter.isAppearanceParameter() && (value == null || parameter.getTypeClass() == value.getClass());
	}
}

File: OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/gpx/GpxDirItem.kt
package net.osmand.shared.gpx

import net.osmand.shared.io.KFile

class GpxDirItem private constructor(
	file: KFile,
	initializeFileParameters: Boolean
) : DataItem(file, initializeFileParameters) {

	constructor(file: KFile) : this(file, true)

	companion object {
		internal fun fromDatabase(file: KFile) = GpxDirItem(file, false)
	}

	override fun isValidValue(parameter: GpxParameter, value: Any?): Boolean {
		if (parameter.isAppearanceParameter()) {
			return value == null || parameter.typeClass == value::class;
		}
		return parameter.isGpxDirParameter() && super.isValidValue(parameter, value);
	}

	override fun getParameters(): Map<GpxParameter, Any?> =
		GpxParameter.getGpxDirParameters().associateWith { parameter -> getParameter(parameter) }
}
